crates/utils/re_video/src/decode/webcodecs.rs

```diff
@@ -190,8 +191,27 @@ impl AsyncDecoder for WebVideoDecoder {
 
         Ok(())
     }
+
+    fn min_num_samples_to_enqueue_ahead(&self) -> usize {
+        // TODO(#8848): For some h264 videos (which??) we need to enqueue more samples, otherwise Safari will not provide us with any frames.
+        // (The same happens with FFmpeg-cli decoder for the affected videos)
+        if self.video_config.is_h264() && *IS_SAFARI {
+            16 // Safari needs more samples queued for h264
+        } else {
+            // No such workaround are needed anywhere else,
+            // GOP boundaries as handled by the video player are enough.
+            0
+        }
+    }
 }
 
+const IS_SAFARI: Lazy<bool> = Lazy::new(|| {
+    web_sys::window()
+        .and_then(|w| w.navigator().user_agent().ok())
+        .map(|ua| ua.contains("Safari"))
+        .unwrap_or(false)
+});
```

Comment:
A lot of modern browser user agents has the string `Safari` in them.

Look at this instead:
https://github.com/rerun-io/rerun/blob/e0c706b01a52fa463f60b4175c4f3b730dbaa002/crates/viewer/re_viewer_context/src/viewer_context.rs#L254-L271

Reply:
thanks! 👀 👀